### ISSUE TITLE
Unable to remove the video component from render tree even after the parent component is killed

### ISSUE THREAD

**Issue by VenkateshSrini** [dotnet/aspnetcore#24537]
Hi,
I'm trying to create a video component with Blazor. I'm using the video.js library for video-related features. The code is in repo https://github.com/VenkateshSrini/BlazorComponents.
Now when I run the code for the first time, The counter page comes up. It has a button called player which when clicked launches the videplayer.razor. This page has a video tag. It displays the vide and I also have a timer on the page that periodically updates the remaining time and duration in localStorage of browser. This is achieved through the Javascript bridge. Then when I again click back the counter page it smoothly transition without error. Now when I repeat the same operation the next time I get a weird error. I'm unable to proceed further I do not understand what is happening. The error is as below

> blazor.webassembly.js:1 crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Cannot read property 'removeChild' of null
      TypeError: Cannot read property 'removeChild' of null
          at e (http://localhost:5000/_framework/blazor.webassembly.js:1:6918)
          at Object.e [as removeLogicalChild] (http://localhost:5000/_framework/blazor.webassembly.js:1:6890)
          at e.applyEdits (http://localhost:5000/_framework/blazor.webassembly.js:1:13649)
          at e.updateComponent (http://localhost:5000/_framework/blazor.webassembly.js:1:12880)
          at Object.t.renderBatch (http://localhost:5000/_framework/blazor.webassembly.js:1:1704)
          at Object.window.Blazor._internal.renderBatch (http://localhost:5000/_framework/blazor.webassembly.js:1:34784)
          at _mono_wasm_invoke_js_unmarshalled (http://localhost:5000/_framework/wasm/dotnet.3.2.0.js:1:172099)
          at wasm_invoke_iiiiii (http://localhost:5000/_framework/wasm/dotnet.wasm:wasm-function[3160]:0x9b33d)
          at icall_trampoline_dispatch (http://localhost:5000/_framework/wasm/dotnet.wasm:wasm-function[5777]:0xfe711)
          at mono_wasm_interp_to_native_trampoline (http://localhost:5000/_framework/wasm/dotnet.wasm:wasm-function[4607]:0xca81d)
Microsoft.JSInterop.JSException: Cannot read property 'removeChild' of null
TypeError: Cannot read property 'removeChild' of null
    at e (http://localhost:5000/_framework/blazor.webassembly.js:1:6918)
    at Object.e [as removeLogicalChild] (http://localhost:5000/_framework/blazor.webassembly.js:1:6890)
    at e.applyEdits (http://localhost:5000/_framework/blazor.webassembly.js:1:13649)
    at e.updateComponent (http://localhost:5000/_framework/blazor.webassembly.js:1:12880)
    at Object.t.renderBatch (http://localhost:5000/_framework/blazor.webassembly.js:1:1704)
    at Object.window.Blazor._internal.renderBatch (http://localhost:5000/_framework/blazor.webassembly.js:1:34784)
    at _mono_wasm_invoke_js_unmarshalled (http://localhost:5000/_framework/wasm/dotnet.3.2.0.js:1:172099)
    at wasm_invoke_iiiiii (http://localhost:5000/_framework/wasm/dotnet.wasm:wasm-function[3160]:0x9b33d)
    at icall_trampoline_dispatch (http://localhost:5000/_framework/wasm/dotnet.wasm:wasm-function[5777]:0xfe711)
    at mono_wasm_interp_to_native_trampoline (http://localhost:5000/_framework/wasm/dotnet.wasm:wasm-function[4607]:0xca81d)
  at Microsoft.JSInterop.WebAssembly.WebAssemblyJSRuntime.InvokeUnmarshalled[T0,T1,T2,TResult] (System.String identifier, T0 arg0, T1 arg1, T2 arg2) <0x2cd31a0 + 0x00046> in <filename unknown>:0 
  at Microsoft.JSInterop.WebAssembly.WebAssemblyJSRuntime.InvokeUnmarshalled[T0,T1,TResult] (System.String identifier, T0 arg0, T1 arg1) <0x2cd30c0 + 0x00014> in <filename unknown>:0 
  at Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer.UpdateDisplayAsync (Microsoft.AspNetCore.Components.RenderTree.RenderBatch& batch) <0x2cd2fc8 + 0x0001e> in <filename unknown>:0 
  at Microsoft.AspNetCore.Components.RenderTree.Renderer.ProcessRenderQueue () <0x2b81948 + 0x000f2> in <filename unknown>:0 

I'm also attaching a zip file that has a vide file (mp4)that shows the steps for reproducing the error. Please help
[X210ipXq97.zip](https://github.com/dotnet/aspnetcore/files/5016651/X210ipXq97.zip)

**Comment by mkArtakMSFT** [comment]
Hi. It looks like this is a question about how to use ASP.NET Core. While we do our best to look through all the issues filed here, we are not a general-purpose forum. To get a faster response we suggest posting your questions to [StackOverflow](https://www.stackoverflow.com) using the **blazor** tag.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @VenkateshSrini.
We do not debug customer applications. If you are able to share a minimalistic application to reproduce the error we may be able to investigate it.

**Comment by VenkateshSrini** [comment]
@mkArtakMSFT,
It is not a customer problem but my own component. It is just using Javascript and Blazor. In fact, I suspect that the issue is coming because we use the Javascript runtime interop. You can see the video that I have uploaded and GitHub code (public url)

**Comment by VenkateshSrini** [comment]
Hi All,

I was able o resolve this issue. I had change the calls to InvokeVoidasync and all the passing of parameter was handled using sessionstorage/localstorage. I was using a timer on the Blazor page. So I had implemented a Dispose method using IDisposable. But to perform the cleanup activity of removing the video control I again had to call Javascript. I could not do that in Dispose because Dispose was a sync method. The GetAwaiter().GetResult(), So  implemented IAsyncDisposable. Then I used the dispose async and dispose method. These things fixed the issue.
